src/Tests/Feature/Web/ThreadStoreTest.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace TeamTeaTime\Forum\Tests\Feature\Web;
+
+use Illuminate\Foundation\Auth\User;
+use Orchestra\Testbench\Factories\UserFactory;
+use TeamTeaTime\Forum\Database\Factories\CategoryFactory;
+use TeamTeaTime\Forum\Models\Category;
+use TeamTeaTime\Forum\Models\Thread;
+use TeamTeaTime\Forum\Models\Post;
+use TeamTeaTime\Forum\Support\Web\Forum;
+use TeamTeaTime\Forum\Tests\TestCase;
+
+class ThreadStoreTest extends TestCase
+{
+    private string $route = 'thread.store';
+
+    private CategoryFactory $categoryFactory;
+    private UserFactory $userFactory;
+
+    private Category $category;
+    private User $user;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+        
+        $this->categoryFactory = CategoryFactory::new();
+        $this->userFactory = UserFactory::new();
+
+        $this->category = $this->categoryFactory->createOne();
+        $this->user = $this->userFactory->createOne();
+    }
+
+    /** @test */
+    public function should_fail_validation_without_content()
+    {
+        $response = $this->actingAs($this->user)
+            ->post(Forum::route($this->route, $this->category), [
+                'title' => "Thread title",
+                'content' => ""
+            ]);
+
+        $response->assertSessionHasErrors();
+    }
+
+    /** @test */
+    public function should_403_when_not_logged_in()
+    {
+        $response = $this->post(Forum::route($this->route, $this->category), []);
+        
+        $response->assertStatus(403);
+    }
+
+    /** @test */
+    public function should_403_when_category_doesnt_accept_threads()
+    {
+        $category = $this->categoryFactory->createOne(['accepts_threads' => 0]);
+        $response = $this->actingAs($this->user)
+            ->post(Forum::route($this->route, $category), []);
+        
+        $response->assertStatus(403);
+    }
+
+    /** @test */
+    public function should_create_a_thread_and_post()
```

Comment:
Why should it? Would something like `should_create_thread_if_first_post` be more descriptive?

Reply:
These tests are specifically for the `thread.store` route, so it's more that the post creation is a side effect of that rather than thread creation being a side effect of creating the first post.

Reply:
Ah ok, fair enough.